dat/writers/create_reference_tables.py

```diff
@@ -14,6 +14,54 @@
 
 spark = configure_spark_with_delta_pip(builder).getOrCreate()
 
+def create_reference_table_1():
+    columns = ["letter", "number", "a_float"]
```

Comment:
These functions violates the single responsibility principle, see https://en.wikipedia.org/wiki/Single-responsibility_principle or "Clean Code" from Uncle Bob.

These function performs: 

1. creating a first dataframe
2. creating path on the file system
3. writing the first dataframe to disk
4. writing the dataframe content as parquet
5. creating a second dataframe
6. writing the second dataframe to disk

First of all, the second activity could be performed as a first one, so the function looks like that

```python

def _ensure_path_exists(table_name):
        table_path =f 'out/tables/generated/{table_name}/expected'
        os.makedirs(table_path)

def create_reference_table_3():
       _ensure_path_exist('reference_table_3')
      df_1 = _create_df_1()
      write_df_as_delta_and_spark(df_1)
      df_2 = _create_df_2()
      write_df_as_delta_and_spark(df_2)
```

Reply:
Now if you take this approach, and you generalize it with respect to the number of transactions, you use an array

```python
def create_reference_table_3():
       _ensure_path_exist('reference_table_3')
      dfs = [ _create_df_1(), _create_df_2() ]
      for df in dfs:
             write_df_as_delta_and_spark(df)
```

If you keep iterating by applying good coding practices such as DRY, single responsibility principle, etc, you end up with something similar to the existing codebase

Reply:
@edmondo1984 - yea, we'll definitely want to make abstractions and clean the code up.  I'm just not sure what the right abstractions are yet.  I wasn't sure how to fit `repartition(1)`, alter table properties, etc. into the current framework.  I am going to try and make a more diverse list of reference tables and then see if we can make abstractions that are sufficiently abstract to handle all the edge cases.

Reply:
I see, yes. I guess one has to extend the object model (there is a RowCollection), you probably need to change that into RowOperation with multiple subclasses (inserts, repertitions)